homeassistant/components/airly/air_quality.py

```diff
@@ -114,25 +112,34 @@ def unique_id(self):
     @property
     def available(self):
         """Return True if entity is available."""
-        return bool(self.data)
+        return self.coordinator.last_update_success
 
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
-        self._attrs[LABEL_ADVICE] = self.data[ATTR_API_ADVICE]
-        self._attrs[LABEL_AQI_LEVEL] = self.data[ATTR_API_CAQI_LEVEL]
-        self._attrs[LABEL_PM_2_5_LIMIT] = self.data[ATTR_API_PM25_LIMIT]
-        self._attrs[LABEL_PM_2_5_PERCENT] = round(self.data[ATTR_API_PM25_PERCENT])
-        self._attrs[LABEL_PM_10_LIMIT] = self.data[ATTR_API_PM10_LIMIT]
-        self._attrs[LABEL_PM_10_PERCENT] = round(self.data[ATTR_API_PM10_PERCENT])
+        self._attrs[LABEL_AQI_DESCRIPTION] = self.coordinator.data[
+            ATTR_API_CAQI_DESCRIPTION
+        ]
+        self._attrs[LABEL_ADVICE] = self.coordinator.data[ATTR_API_ADVICE]
+        self._attrs[LABEL_AQI_LEVEL] = self.coordinator.data[ATTR_API_CAQI_LEVEL]
+        self._attrs[LABEL_PM_2_5_LIMIT] = self.coordinator.data[ATTR_API_PM25_LIMIT]
+        self._attrs[LABEL_PM_2_5_PERCENT] = round(
+            self.coordinator.data[ATTR_API_PM25_PERCENT]
+        )
+        self._attrs[LABEL_PM_10_LIMIT] = self.coordinator.data[ATTR_API_PM10_LIMIT]
+        self._attrs[LABEL_PM_10_PERCENT] = round(
+            self.coordinator.data[ATTR_API_PM10_PERCENT]
+        )
```

Comment:
I am confused why you store this dictionary as an instance variable? why are you not returning it directly.

```python
data = self.coordinator.data

return {
  LABEL_AQI_DESCRIPTION: data[ATTR_API_CAQI_DESCRIPTION],
  LABEL_ADVICE: data[ATTR_API_DEVICE],
  # …
}
```

Reply:
Fixed.